File: problems/253/solution.js
/**
 * @param {number[][]} intervals
 * @return {number}
 */
var minMeetingRooms = function(intervals) {
    const starts = intervals.map(intervals=>intervals[0]);
    const ends = intervals.map(intervals=>intervals[1]);
    sortNumbers(starts);
    sortNumbers(ends);
    let [i, j, counter, max] = [0,0,0,0]
    const size = starts.length;
    while((i < size) && (j < size)){
      while(starts[i] < ends[j]){
        max = Math.max(++counter, max);
        i++;
      }
      while(starts[i] >= ends[j]){
        counter--;
        j++;
      }
    }
    return max;
};

const sortNumbers = nums =>{
  return nums.sort((a, b)=>a - b)
}

File: problems/253/solution.cpp
class Solution {
public:
    int minMeetingRooms(vector<vector<int>>& intervals) {
        int ans = 0, cur = 0;
        vector<pair<int, int>> line;
        
        for (const vector<int>& interval: intervals) {
            line.push_back({interval[0], 1});
            line.push_back({interval[1], -1});
        }
        
        sort(begin(line), end(line));
        
        for (auto [ignore, change]: line) {
            cur += change;
            ans = max(ans, cur);
        }
        
        return ans;
    }
};